### PR TITLE
Improve rendering performance of the activity feed

### PR DESCRIPTION
**Pull request by rafenden**
https://trello.com/c/LSCn0Yvf/1295-stress-test-activity-feed-performance-with-high-numbers-of-interactions

### CODE REVIEW

File: src/activity-feed/ActivityFeed.test.jsx [hunk 1]
```diff
@@ -2,63 +2,134 @@ import React from 'react'
...
+    super(props)
+    this.state = {
+      activities: [],
+    }
+  }
+
+  onLoadMore = () => {
+    const { activities } = this.state
+    const generatedActivities = []
+    for (let i = activities.length; i < activities.length + 10; i += 1) {
```
Comment (yeahfro) [inline src/activity-feed/ActivityFeed.test.jsx]: Can this just be `map`?

File: src/activity-feed/ActivityFeed.test.jsx [hunk 2]
```diff
@@ -2,63 +2,134 @@ import React from 'react'
...
-      .create(
-        <ActivityFeed
-          isLoading={true}
+  describe('when there are more activities to load', () => {
+    test('should render with "Load more" link', () => {
+      const tree = renderer
+        .create(<ActivityFeed
+          totalActivities={20}
+          activities={
+            Array.from({ length: 20 }, (e, i) => ({
```
Comment (yeahfro) [inline src/activity-feed/ActivityFeed.test.jsx]: This is a different method for generating activity fixtures

Comment (rafenden) [reply src/activity-feed/ActivityFeed.test.jsx]: Can you clarify please?

File: src/activity-feed/ActivityFeed.test.jsx (hunk 1 above)
Comment (rafenden) [reply src/activity-feed/ActivityFeed.test.jsx]: Not sure which one is cleaner:
```
Array.from({ length: 20 }).map(...)
```

or 
```
for (let i = activities.length; i < activities.length + 10; i += 1)
```

File: src/activity-feed/ActivityFeed.test.jsx [hunk 3]
```diff
@@ -1,64 +1,131 @@
...
+    super(props)
+    this.state = {
+      activities: [],
+    }
+  }
+
+  onLoadMore = () => {
+    this.setState((prevState) => {
+      const { activities } = prevState
+      const allActivities = activities.concat(generateActivities(10))
```
Comment (yeahfro) [inline src/activity-feed/ActivityFeed.test.jsx]: 👍

File: src/activity-feed/ActivityFeed.test.jsx [hunk 4]
```diff
@@ -1,64 +1,131 @@
...
-    const tree = renderer
-      .create(
-        <ActivityFeed
-          isLoading={true}
+  describe('when there are more activities to load', () => {
+    test('should render with "Load more" link', () => {
+      const tree = renderer
+        .create(<ActivityFeed
+          totalActivities={20}
+          activities={generateActivities(20)}
```
Comment (yeahfro) [inline src/activity-feed/ActivityFeed.test.jsx]: 👍

**Pull request merged**